app/src/main/java/com/m2049r/xmrwallet/util/LocaleHelper.java

```diff
@@ -0,0 +1,85 @@
+package com.m2049r.xmrwallet.util;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.res.Configuration;
+import android.preference.PreferenceManager;
+
+import com.m2049r.xmrwallet.R;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Locale;
+
+public class LocaleHelper {
+    private static final String PREFERRED_LOCALE_KEY = "preferred_locale";
+    private static Locale SYSTEM_DEFAULT_LOCALE = Locale.getDefault();
+
+    public static ArrayList<Locale> getAvailableLocales(Context context) {
+        ArrayList<Locale> locales = new ArrayList<>();
+        HashSet<String> localizedStrings = new HashSet<>();
+        int compareResId = R.string.menu_about;
+
+        // Application default locale
+        locales.add(Locale.ENGLISH);
+        localizedStrings.add(getLocaleString(context, Locale.ENGLISH, compareResId));
+
+        // Locale with region
+        locales.add(Locale.CHINA);
+        localizedStrings.add(getLocaleString(context, Locale.CHINA, compareResId));
+
+        locales.add(Locale.TAIWAN);
+        localizedStrings.add(getLocaleString(context, Locale.TAIWAN, compareResId));
```

Comment:
why is it necessary to add these three manually? why are they special?

Reply:
When enumerating system supported locales and comparing string, system will fallback to other region which use the same language if the locale is not translated.

For instance, zh-CN, zh-HK and zh-TW are used by China, Hong Kong and Taiwan respectively.
If we don't add locale manually, "Chinese (Hong Kong)" will be listed because system returns translated string from zh-TW.

So, i add them manually to inform user that this is translated for the specific region and there may be different translated terms corresponding to the same English.
In short, just for more accurate translated terms.